src/sub.cpp

```diff
@@ -59,10 +59,7 @@ int zmq::sub_t::xsetsockopt (int option_,
     int rc = msg.init_size (optvallen_ + 1);
     errno_assert (rc == 0);
     unsigned char *data = (unsigned char *) msg.data ();
-    if (option_ == ZMQ_SUBSCRIBE)
-        *data = 1;
-    else if (option_ == ZMQ_UNSUBSCRIBE)
-        *data = 0;
+    *data = (option_ == ZMQ_SUBSCRIBE);
```

Comment:
Can't remember what the spec says right now - are you sure that on every platform, every architecture, every compiler and so on "true" is going to evaluate to the exactly 0x1? Otherwise things are going to break

Reply:
Yes: section 4.5 paragraph 4 of the C++ standard says: "An rvalue of type bool can be converted to an rvalue of type int, with false becoming zero and true becoming one."

Reply:
Grand, thanks for checking.